src/HotChocolate/Core/src/Types/Types/InputParser.cs

```diff
@@ -698,36 +643,57 @@ private object DeserializeObject(object resultValue, InputObjectType type, Path
                 : value;
         }
 
-        try
+        return ParseAndFormatAndConvertLiteral(field.DefaultValue, path, stack, false, field, field.IsOptional);
+    }
+
+    private object? ParseAndFormatAndConvertLiteral(
+        IValueNode literal,
+        Path fieldPath,
+        int stack,
+        bool defaults,
+        IInputValueInfo field,
+        bool isOptional)
+    {
+        var value = ParseLiteralInternal(
+            literal,
+            field.Type,
+            fieldPath,
+            stack,
+            defaults,
+            field);
+        return FormatAndConvertValue(field, value, fieldPath, isOptional);
+    }
+
+    private object? FormatAndConvertValue(
+        IInputValueInfo inputValueInfo,
+        object? value,
+        Path path,
+        bool isOptional,
+        Type? requestedType = null)
+    {
+        value = FormatValue(inputValueInfo, value);
+        value = ConvertValue(requestedType ?? inputValueInfo.RuntimeType, value, out var conversionException);
+        if (conversionException != null)
         {
-            value = ParseLiteralInternal(
-                field.DefaultValue,
-                field.Type,
-                path,
-                stack,
-                false,
-                field);
+            throw InvalidTypeConversion(inputValueInfo.Type, path, inputValueInfo, conversionException);
         }
-        catch (SerializationException ex)
+
+        if (isOptional)
         {
-            throw new SerializationException(ex.Errors[0].WithPath(path), ex.Type, path);
+            value = new Optional(value, true);
```

Comment:
I think the `true` is wrong here.
If you look at the diff the `hasValue` constructor argument is being set to `false` in the removed portion of the code in the `CreateDefaultValue(IInputField)` path.
Maybe you can pass it through as an arg.